gulpfile.js

```diff
@@ -1106,6 +1121,67 @@ function compileJs(srcDir, srcFilename, destDir, options) {
   }
 }
 
+/**
+ * Given a file path `foo/bar.js`, transpiles the TypeScript entry point of
+ * the same name `foo/bar.ts` and all direct and indirect TypeScript imports.
+ *
+ * @param {string} srcDir
+ * @param {string} srcFilename
+ */
+function transpileTs(srcDir, srcFilename) {
+  const startTime = Date.now();
+
+  const tsEntry = path.join(srcDir, srcFilename).replace('.js', '.ts');
+  const tsConfig = ts.convertCompilerOptionsFromJson({
```

Comment:
Does this imply a specific convention for the location of a `tsconfig.json` file relative to the extension?

Reply:
Makes sense in most cases. However, for TS there are different configurations for code in a WorkerGlobalScope versus the UI thread. This might cause a few issues in those scenarios.